src/andromede/simulation/benders_decomposed.py

```diff
@@ -208,54 +209,76 @@ def run(
 
 
 def build_benders_decomposed_problem(
-    network_on_tree: Dict[TreeNode, Network],
+    decision_tree_root: DecisionTreeNode,
     database: DataBase,
-    configured_tree: ConfiguredTree,
     *,
-    decision_coupling_model: Optional[Model] = None,
     border_management: BlockBorderManagement = BlockBorderManagement.CYCLE,
     solver_id: str = "GLOP",
+    coupling_network: Network = Network(""),
 ) -> BendersDecomposedProblem:
     """
     Entry point to build the xpansion problem for a time period
 
     Returns a Benders Decomposed problem
     """
 
-    master_network = create_master_network(network_on_tree, decision_coupling_model)
+    if not decision_tree_root.is_leaves_prob_sum_one():
+        raise RuntimeError("Decision tree must have leaves' probability sum equal one!")
+
+    null_time_block = TimeBlock(
+        0, [0]
+    )  # Not necessary for master, but list must be non-empty
+    null_scenario = 0  # Not necessary for master
 
-    # Benders Decomposed Master Problem
-    master = build_problem(
-        master_network,
+    coupler = build_problem(
+        coupling_network,
         database,
-        null_time_block := TimeBlock(  # Not necessary for master, but list must be non-empty
-            0, [0]
-        ),
-        null_scenario := 0,  # Not necessary for master
-        problem_name="master",
-        border_management=border_management,
+        null_time_block,
+        null_scenario,
+        problem_name="coupler",
         solver_id=solver_id,
-        problem_strategy=InvestmentProblemStrategy(),
+        build_strategy=InvestmentProblemStrategy(),
+        risk_strategy=ExpectedValue(0.0),
     )
 
-    # Benders Decomposed Sub-problems
-    subproblems = []
-    for (
-        tree_node,
-        time_scenario_config,
-    ) in configured_tree.node_to_config.items():
-        for block in time_scenario_config.blocks:
-            # Xpansion Sub-problems
+    masters = []  # Benders Decomposed Master Problem
+    subproblems = []  # Benders Decomposed Sub-problems
+
+    for tree_node in decision_tree_root.traverse():
+        suffix = f"_{tree_node.id}" if decision_tree_root.size > 1 else ""
```

Comment:
In this case, it would represent the current behavior, so no pathway.
It would be unnecessary to add a suffix to master and sub-problems since they all belong to the same node